pkg/cmd/experimental/syncgroups/ad/ldapinterface.go

```diff
@@ -150,3 +150,20 @@ func (e *ADLDAPInterface) requiredUserAttributes() []string {
 
 	return allAttributes.List()
 }
+
+// Exists determines if a group idenified with it's LDAP group UID exists on the LDAP server
+func (e *ADLDAPInterface) Exists(ldapGrouUID string) (bool, error) {
```

Comment:
Add unit tests for all these methods.  Every `Exists` method and be sure to check each branch.